modeling-cmds/src/def_enum.rs

```diff
@@ -1557,6 +1557,47 @@ define_modeling_cmd_enum! {
             /// List of transforms to be applied to the object.
             pub transforms: Vec<ComponentTransform>,
         }
+
+        /// Create a new solid from combining other smaller solids.
+        /// In other words, every part of the input solids will be included in the output solid.
+        #[derive(
+            Clone, Debug, Deserialize, PartialEq, JsonSchema, Serialize, ModelingCmdVariant,
+        )]
+        #[cfg_attr(feature = "ts-rs", derive(ts_rs::TS))]
+        #[cfg_attr(feature = "ts-rs", ts(export_to = "ModelingCmd.ts"))]
+        pub struct BooleanUnion
+        {
+            /// Which solids to union together
+            pub solid_ids: Vec<ModelingCmdId>,
+        }
+
+        /// Create a new solid from intersecting several other solids.
+        /// In other words, the part of the input solids where they all overlap will be the output solid.
+        #[derive(
+            Clone, Debug, Deserialize, PartialEq, JsonSchema, Serialize, ModelingCmdVariant,
+        )]
+        #[cfg_attr(feature = "ts-rs", derive(ts_rs::TS))]
+        #[cfg_attr(feature = "ts-rs", ts(export_to = "ModelingCmd.ts"))]
+        pub struct BooleanIntersection
+        {
+            /// Which solids to intersect together
+            pub solid_ids: Vec<ModelingCmdId>,
+        }
+
+        /// Create a new solid from subtracting several other solids.
+        /// The 'target' is what will be cut from.
+        /// The 'tool' is what will be cut out from 'target'.
```

Comment:
I don't see the `tool` parameter.

That said, I read the comments about why it was proposed, and I personally don't think we should have it. It makes subtraction different from the other ops, meaning there will need to be a lot more busywork code to convert and special-case subtraction. I don't think the argument that it's more intuitive for MEs applies since this is the API. Software engineers are the ones working with the API. ME ergonomics should be in the UI or KCL. KCL can have a `tools` keyword param that does the mapping.

Reply:
First of all, subtraction is necessarily different from union/intersection. Because it has to use two different types of solid, that aren't interchangeable (the LHS and RHS of the difference operator).

So we will need two fields in this struct. They could both be single UUIDs, but they may as well be vectors as it's more flexible.

Now, if we use two different vectors of solid IDs, they'll need different names to distinguish which is LHS and RHS of the subtraction. I think tool and target are decent enough names for those.